Traceback (most recent call last):
  File "C:\nb\projects\blog\.venv\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\nb\projects\blog\.venv\Lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\nb\projects\blog\.venv\Lib\site-packages\jupyter_core\utils\__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\asyncio\base_events.py", line 650, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\contextlib.py", line 222, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "C:\nb\projects\blog\.venv\Lib\site-packages\nbclient\client.py", line 650, in async_setup_kernel
    yield
  File "C:\nb\projects\blog\.venv\Lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "C:\nb\projects\blog\.venv\Lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\nb\projects\blog\.venv\Lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
img_url = "https://github.com/scikit-image/scikit-image/blob/main/skimage/data/astronaut.png?raw=true"
img = mpimg.imread(img_url)
plt.imshow(img)
plt.show()
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mValueError[0m                                Traceback (most recent call last)
Cell [1;32mIn[2], line 2[0m
[0;32m      1[0m img_url [38;5;241m=[39m [38;5;124m"[39m[38;5;124mhttps://github.com/scikit-image/scikit-image/blob/main/skimage/data/astronaut.png?raw=true[39m[38;5;124m"[39m
[1;32m----> 2[0m img [38;5;241m=[39m [43mmpimg[49m[38;5;241;43m.[39;49m[43mimread[49m[43m([49m[43mimg_url[49m[43m)[49m
[0;32m      3[0m plt[38;5;241m.[39mimshow(img)
[0;32m      4[0m plt[38;5;241m.[39mshow()

File [1;32mC:\nb\projects\blog\.venv\Lib\site-packages\matplotlib\image.py:1558[0m, in [0;36mimread[1;34m(fname, format)[0m
[0;32m   1554[0m img_open [38;5;241m=[39m (
[0;32m   1555[0m     PIL[38;5;241m.[39mPngImagePlugin[38;5;241m.[39mPngImageFile [38;5;28;01mif[39;00m ext [38;5;241m==[39m [38;5;124m'[39m[38;5;124mpng[39m[38;5;124m'[39m [38;5;28;01melse[39;00m PIL[38;5;241m.[39mImage[38;5;241m.[39mopen)
[0;32m   1556[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(fname, [38;5;28mstr[39m) [38;5;129;01mand[39;00m [38;5;28mlen[39m(parse[38;5;241m.[39murlparse(fname)[38;5;241m.[39mscheme) [38;5;241m>[39m [38;5;241m1[39m:
[0;32m   1557[0m     [38;5;66;03m# Pillow doesn't handle URLs directly.[39;00m
[1;32m-> 1558[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[0;32m   1559[0m         [38;5;124m"[39m[38;5;124mPlease open the URL for reading and pass the [39m[38;5;124m"[39m
[0;32m   1560[0m         [38;5;124m"[39m[38;5;124mresult to Pillow, e.g. with [39m[38;5;124m"[39m
[0;32m   1561[0m         [38;5;124m"[39m[38;5;124m``np.array(PIL.Image.open(urllib.request.urlopen(url)))``.[39m[38;5;124m"[39m
[0;32m   1562[0m         )
[0;32m   1563[0m [38;5;28;01mwith[39;00m img_open(fname) [38;5;28;01mas[39;00m image:
[0;32m   1564[0m     [38;5;28;01mreturn[39;00m (_pil_png_to_float_array(image)
[0;32m   1565[0m             [38;5;28;01mif[39;00m [38;5;28misinstance[39m(image, PIL[38;5;241m.[39mPngImagePlugin[38;5;241m.[39mPngImageFile) [38;5;28;01melse[39;00m
[0;32m   1566[0m             pil_to_array(image))

[1;31mValueError[0m: Please open the URL for reading and pass the result to Pillow, e.g. with ``np.array(PIL.Image.open(urllib.request.urlopen(url)))``.

